enhancements/sandboxed-containers/sandboxed-containers-tech-preview.md

```diff
@@ -0,0 +1,620 @@
+   Kata Containers - as part of the RHCOS Extensions, for all users relying
+   on extensions; ~150MB installed for those using the Kata Containers
+   extension.
+
+In case this is the route to be taken, work on the following components will
+have to done:
+* OpenShift: Create a PR for https://github.com/openshift/os, adding
+  sandboxed containers as a new extension (and, most likely, reverting the
+  addition of the `qemu-kiwi` extension).
+* OpenShift: Follow up the PR opened for the `machine-config-operator` to
+  have `sandboxed-containers` added as an extension.  For reference, see:
+  https://github.com/openshift/machine-config-operator/pull/2456
+* sandboxed containers operator: Adapt our operator to deploy those RPMs
+  using the CoreOS Extensions.
+  
+#### Use RHCOS extensions for QEMU and its dependencies, copy Kata Containers files onto the host
+
+In this option we completely rely on RHCOS Extensions for the QEMU and its
+dependencies, but have to rely on our own provided payload containing the
+binaries provided by kata-containers RPM, which would be copied to the RHCOS
+node by the sandboxed containers operator.
+
+**Pros:**
+1. The most size costly RPMs can be installed only by the users who would be
+   interested in using the Sandboxed Containers runtimes.
+2. There's no extra work related to properly loading QEMU dependencies, as
+   all the packages would be in installed in the same PATH its build was
+   intent for.
+3. QEMU RPMs have been tested / validated using an identical environment.
+4. No additional work with regarding to re-packaging those in another form
+   other than the RPMs which is the way they are consumed by other Layered
+   Products.
+5. We rely on an already existing mechanism to install part of the RPMs, QEMU
+   and its dependencies, which are not part of the base image.
+
+**Cons:**
+1. 5.5MB extra to be download - coming from QEMU and its dependencies - as
+   part of the RHCOS Extensions, for all users relying on extensions; ~20MB
+   installed for those using the QEMU extension.
+2. 130MB extra - coming from Kata Containers binaries - copied onto the host,
+   for those who'd be using the Sandboxed Containers,
+3. The sandboxed containers operator still does a work that, in theory,
+   should be restricted to the MCO.
+
+In case this is the route to be taken, work on the following components will
+have to done:
+* OpenShift: Follow up the PR opened for the `machine-config-operator` to
+  have `sandboxed-containers` added as an extension.  For reference, see:
+  https://github.com/openshift/machine-config-operator/pull/2456
+* sandboxed containers operator: Adapt our operator to deploy those RPMs
+  using the CoreOS Extensions.
+* sandboxed containers operator: Adapt our RPMs to be installed in
+  `/opt/sandboxed-containers/`
+* sandboxed containers operator: Adapt our operator to copy the binaries
+  on to the RHCOS node.
+
```

Comment:
We need a section to cover how we will restrict access to kata runtime class. Mention that SCC is being discussed but we may end up using a custom solution for kata.

Reply:
@haircommander can you help @fidencio with this section?

Reply:
I think that can be left out of the proposal and added later when there is consensus on how that would be implemented given upstream activities. (We can later add the section in a post PR with the details once they are there?)